nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3DirectBatchHandler.java

```diff
@@ -214,11 +218,19 @@ public boolean completeUpload(String batchId, String fileIndex, BatchFileInfo fi
         String mimeType = metadata.getContentType();
         String encoding = metadata.getContentEncoding();
 
+        // server-side encryption
+        String targetSSEAlgorithm;
+        if (useServerSideEncryption) { // TODO KMS
+            targetSSEAlgorithm = ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION;
+        } else {
+            targetSSEAlgorithm = null;
+        }
+
         ObjectMetadata newMetadata;
         if (metadata.getContentLength() > lowerThresholdToUseMultipartCopy()) {
-            newMetadata = S3Utils.copyFileMultipart(amazonS3, metadata, bucket, fileKey, bucket, newFileKey, true);
+            newMetadata = S3Utils.copyFileMultipart(amazonS3, metadata, bucket, fileKey, bucket, newFileKey, targetSSEAlgorithm, true);
         } else {
-            newMetadata = S3Utils.copyFile(amazonS3, metadata, bucket, fileKey, bucket, newFileKey, true);
+            newMetadata = S3Utils.copyFile(amazonS3, metadata, bucket, fileKey, bucket, newFileKey, targetSSEAlgorithm, true);
             boolean isMultipartUpload = REGEX_MULTIPART_ETAG.matcher(etag).find();
```

Comment:
as seen yesterday together we can remove the block if(isMultipartUpload())

Reply:
I'd rather do non-trivial changes in a separate issue